packages/plugins/src/hardhat/simulate.ts

```diff
@@ -337,14 +255,58 @@ export const fetchTransactionResponses = async (
  * be brittle because Hardhat could change their internal functionality in a future minor or patch
  * version.
  */
-export const simulateDeploymentSubtask = async (
-  taskArgs: simulateDeploymentSubtaskArgs,
-  hre: any
+export const simulateDeployment = async (
+  deploymentConfig: DeploymentConfig,
+  forkUrl: string,
+  chainId: string,
+  blockGasLimit: string,
+  blockNumber: string | undefined,
+  cachePath: string
 ): Promise<{ transactions: SimulationTransactions }> => {
-  // Wrap the Hardhat provider with a Proxy, which implements retry and timeout logic.
-  const provider = createHardhatEthersProviderProxy(hre.ethers.provider)
+  const config: HardhatConfig = resolveConfig('', {
+    paths: {
+      cache: join(process.cwd(), cachePath, '/sphinx'),
```

Comment:
Previously the cached files would have been written to a `cache` folder that is located with the hardhat project (inside of the `node_modules` folder where the `hardhatRunner` was), but since we now don't use a hardhat project, it will write all the files to a `cache` folder in the users project by default. To keep things simple for the user, I updated it so that the files are written into their Foundry cache folder instead.